libs/services/src/lib/ftu-data/ftu-data.impl.ts

```diff
@@ -0,0 +1,371 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable, InjectionToken, inject } from '@angular/core';
+import { createLinkId } from '@hra-ui/cdk/state';
+import { Observable, firstValueFrom, from, map, of } from 'rxjs';
+import { z } from 'zod';
+import { IRI, Iri, URL, Url } from '../shared/common.model';
+import {
+  CellSummary,
+  DataFileReference,
+  IllustrationMappingItem,
+  SourceReference,
+  TissueLibrary,
+} from './ftu-data.model';
+import { FtuDataService } from './ftu-data.service';
+
+/** Endpoints for Injecting input path */
+export interface FtuDataImplEndpoints {
+  /** Endpoint for File having Cell illustrations data */
+  illustrations: Url;
+  /** Endpoint for File having Cell Summaries data */
+  summaries: Url;
+  /** Endpoint for File having Cell Source References data */
+  datasets: Url;
+}
+/** Constant  to read the endpoints */
+export const FTU_DATA_IMPL_ENDPOINTS = new InjectionToken<FtuDataImplEndpoints>('Endpoints');
+
+/** Input to different file formats supported */
+export const FTU_DATA_IMPL_FILE_FORMAT_MAPPING = new InjectionToken<Record<string, string>>('Mapping of file formats', {
+  providedIn: 'root',
+  factory: () => ({
+    'image/svg+xml': 'svg',
+    'image/png': 'png',
+    'application/pdf': 'pdf',
```

Comment:
I know this is weird, but application/pdf in this context is actually an adobe illustrator file (ai)

Reply:
Okay sure. WIll refactor that